src/base/torrentfileswatcher.cpp

```diff
@@ -470,6 +471,14 @@ void TorrentFilesWatcher::Worker::removeWatchedFolder(const QString &path)
         m_retryTorrentTimer->stop();
 }
 
+void TorrentFilesWatcher::Worker::scheduleWatchedFolderProcessing(const QString &path)
+{
+    QTimer::singleShot(2000, this, [this, path]()
+    {
+        processWatchedFolder(path);
```

Comment:
Is it possible that a torrent is added just before the folder is processed? or the singleshot will be rescheduled?
If the answer is not, then this method is not bulletproof enough. Better scan the folder and take a list of files that matches the file extensions and process them later.

Reply:
I just fixed regression (i.e. returned back the old logic). I don't want to implement some other algorithms now.

Reply:
>I don't want to implement some other algorithms now.

I understand but this isn't something new, this is in fact the true old logic:
https://github.com/qbittorrent/qBittorrent/commit/2993fdb16954a9e3301efb7f9171ae1f03ab7743#diff-ac231f3afbec8fdc3eaec22e855c177bd2264f5ea2d070767495f61abc77f9c6L169-L186 (filesystemwatcher.cpp)

Reply:
well... forget my previous post.. maybe I'll improve it later.

Reply:
IMO, if you want to make reliable delay, you should check for file creation time.